src/pages/Reader.tsx

```diff
@@ -29,47 +32,78 @@ class Reader extends React.Component<any, State> { // fix typing up
       images: [],
       pageCount: 0,
       mangafile: this.props.match.params.mangafile,
-      chapter: this.props.match.params.chapter,
-      page: this.props.match.params.page,
+      chapter: this.props.match.params.chapter-1,
+      chapters: [],
+      page: this.props.match.params.page-1,
+      pages: [],
       readerSettings: {
         readerMode: ReaderMode.longStrip,
-        imageSizing: ImageSizing.fitHeight
-      }
+        imageSizing: ImageSizing.fitHeight,
+        colourTheme: ColourTheme.light,
+      },
     };
   }
 
 
   queryManga = async () => {
-    const qs = queryString.parse(this.props.location.search);
-
-    const zipInfo = await fetchZip(`https://files.karatsubascans.com/${qs.mangafile}`);
-
-    // const inflatedImages: string[] = [];
-    // for await (const deflated of zipInfo) {
-    //   const inflated = await extractZip(deflated.contents);
-    //   inflatedImages.push(inflated.toString('base64'));
-    // }
+    // const qs = queryString.parse(this.props.location.search);
+
+    // const zipInfo = await fetchZip(`https://files.karatsubascans.com/${qs.mangafile}`);
+
+    // // const inflatedImages: string[] = [];
+    // // for await (const deflated of zipInfo) {
+    // //   const inflated = await extractZip(deflated.contents);
+    // //   inflatedImages.push(inflated.toString('base64'));
+    // // }
+
+    // this.setState(curState => ({
+    //   ...curState,
+    //   zipped: zipInfo,
+    //   images: zipInfo.map(zip => null),
+    //   pageCount: zipInfo.length
+    // }));
+
+    
+
+    const chapters = await getChapters(this.state.mangafile);
+    const pages = await getPages(this.state.mangafile, chapters[this.state.chapter].name);
+    this.setState({
+      chapters,
+      pages
+    });
 
-    this.setState(curState => ({
-      ...curState,
-      zipped: zipInfo,
-      images: zipInfo.map(zip => null),
-      pageCount: zipInfo.length
-    }));
   }
 
-  loadPage = async (pageNumber: number) => {
+  loadPage = async (pageNumber = -1) => {
     /* TODO check if page is valid */
 
-    const inflated = await extractZip(this.state.zipped[pageNumber].contents);
-    const rawImg = inflated.toString('base64');
-
-    this.setState(curState => {
-      return {
-        ...curState,
-        images: this.state.images.splice(pageNumber, 1, rawImg)
+    // const inflated = await extractZip(this.state.zipped[pageNumber].contents);
+    // const rawImg = inflated.toString('base64');
+
+    // this.setState(curState => {
+    //   return {
+    //     ...curState,
+    //     images: this.state.images.splice(pageNumber, 1, rawImg)
+    //   }
+    // });
+
+    if (pageNumber === -1) {
+      console.log(this.state.chapters, 'chapters');
+      console.log(this.state.pages, 'pages');
+      // load everything
+      const images = [];
+      for (let i = 0; i < this.state.pages.length; i += 1) {
+        images.push(`https://files.karatsubascans.com/${this.state.mangafile}/jpg/`+

```

Comment:
maybe move the `https://file/karatsubascans.com` part to its own API_URL config variable